From 31c5010604979ceddebb3ae67965c240a7f5c16e Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Fri, 10 Jun 2011 09:27:24 +0800
Subject: [PATCH] ENGR00151370 ahci: re-add the temperature monitor sysfs

Add the AHCI temperature monitor sysfs API that removed
accidently in ENGR00144104

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/plat-mxc/ahci_sata.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-mxc/ahci_sata.c b/arch/arm/plat-mxc/ahci_sata.c
index 9fc552c..2f76ff8 100644
--- a/arch/arm/plat-mxc/ahci_sata.c
+++ b/arch/arm/plat-mxc/ahci_sata.c
@@ -495,6 +495,10 @@ static int sata_init(struct device *dev)
 		}
 	}
 
+	/* Add the temperature monitor */
+	ret = sysfs_create_group(&dev->kobj, &fsl_sata_ahci_group);
+	if (ret)
+		sysfs_remove_group(&dev->kobj, &fsl_sata_ahci_group);
 	iounmap(mmio);
 	return ret;
 
@@ -514,6 +518,7 @@ put_sata_clk:
 
 static void sata_exit(struct device *dev)
 {
+	sysfs_remove_group(&dev->kobj, &fsl_sata_ahci_group);
 	if (machine_is_mx53_smd() || machine_is_mx53_loco()
 			|| board_is_mx53_ard_b()) {
 		/* FSL IMX AHCI SATA uses the internal usb phy1 clk */
-- 
1.5.4.4

